compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -975,15 +993,15 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         path: Vec<PathElem>,
         ref_tracking: Option<&mut RefTracking<MPlaceTy<'tcx, M::Provenance>, Vec<PathElem>>>,
         ctfe_mode: Option<CtfeValidationMode>,
-    ) -> InterpResult<'tcx> {
+    ) -> InterpResult<'tcx, Vec<PathElem>> {
```

Comment:
What is the meaning of the returned path? Elements get pushed and popped during visiting; if it's just the final `visitor.path` shouldn't that be equal to the initial path?

Reply:
this is just to avoid a clone in the happy path

Reply:
That needs a comment, then.